File: wasm/bmxt-core/src/cmd/close.rs
use crate::meta::Cmd;
use crate::model::{DispatchJson, Effect};

pub const CMD: Cmd = Cmd {
    name: "close",
    aliases: &["c"],
    usage_primary: "close <tabId>",
    man: &[
        "NAME",
        "  close, c - close a tab",
        "",
        "SYNOPSIS",
        "  close <tabId>",
        "  c <tabId>",
    ],
};

pub fn run(args: &[String]) -> DispatchJson {
    let Some(id) = args.get(1).and_then(|s| s.parse::<i32>().ok()) else {
        return DispatchJson::lines(vec!["usage: close <tabId>".to_string()]);
    };
    DispatchJson::effects(vec![Effect::CloseTab { tab_id: id }])
}

File: lib/features/bmxt-core/cmd/close.ts
import { getRunLocale } from "../../setting/i18n/run-locale"
import { tCmd } from "../../setting/i18n/ns/cmd"
import type { CmdMeta } from "../types"
import { effectsDispatch, linesDispatch } from "../types"

export const CMD: CmdMeta = {
  name: "close",
  aliases: ["c"],
  usagePrimary: "close <tabId>"
}

export function run(args: string[]) {
  const locale = getRunLocale()
  const id = Number.parseInt(args[1] ?? "", 10)
  if (!Number.isFinite(id)) {
    return linesDispatch([tCmd("cmd.close.usage", locale)])
  }
  return effectsDispatch([{ kind: "close_tab", tab_id: id }])
}
